packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1394,35 +1394,39 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
 
   void _handleMoveCursorForwardByCharacter(bool extentSelection) {
     assert(selection != null);
-    final int extentOffset = _textPainter.getOffsetAfter(selection.extentOffset);
+    final int? extentOffset = _textPainter.getOffsetAfter(selection!.extentOffset);
     if (extentOffset == null)
       return;
-    final int baseOffset = !extentSelection ? extentOffset : selection.baseOffset;
+    final int baseOffset = !extentSelection ? extentOffset : selection!.baseOffset;
     _handleSelectionChange(
       TextSelection(baseOffset: baseOffset, extentOffset: extentOffset), SelectionChangedCause.keyboard,
     );
   }
 
   void _handleMoveCursorBackwardByCharacter(bool extentSelection) {
     assert(selection != null);
-    final int extentOffset = _textPainter.getOffsetBefore(selection.extentOffset);
+    final int? extentOffset = _textPainter.getOffsetBefore(selection!.extentOffset);
     if (extentOffset == null)
       return;
-    final int baseOffset = !extentSelection ? extentOffset : selection.baseOffset;
+    final int baseOffset = !extentSelection ? extentOffset : selection!.baseOffset;
     _handleSelectionChange(
       TextSelection(baseOffset: baseOffset, extentOffset: extentOffset), SelectionChangedCause.keyboard,
     );
   }
 
   void _handleMoveCursorForwardByWord(bool extentSelection) {
     assert(selection != null);
-    final TextRange currentWord = _textPainter.getWordBoundary(selection.extent);
-    if (currentWord == null)
+    final TextRange currentWord = _textPainter.getWordBoundary(selection!.extent);
+    // `currentWord` has a non-nullable return type, but might be null when
```

Comment:
as far as i can tell, currentWord actually can never return null. if i'm right, we should just remove the check for currentWord being null.

Reply:
Done